### PR TITLE
Respect robots txt

### LINKED ISSUE #42: Respect robots.txt when crawling when set as True
- Option to set `respect_robots_txt` (Default should be `True` because of legal obligation in some jurisdictions)
- Fetch and Parse robots.txt (`urllib.robotparser` will helps in parsing robots.txt)
- Create crawl rule per domain
- Check URL permissions before crawling a URL
- Make sure it works when concurrent workers are fetching different domains
- Use the rules provided in robots.txt to fetch. (eg: use robots.txt `crawl-delay` if present. Check the rule before crawling a path)

### PR DESCRIPTION
**Pull request by Mews**
Closes #42 

**Changes**
- Added a new `networking.robots_txt` submodule;
- Separated some of the crawler logic to separate functions to avoid pylint's `too-many-branches`;
- `Spider.crawl` now checks whether scraping a website is allowed and also respects specified crawl delay;
- Added new tests for the `robots_txt` submodule and for the `respect_robots_txt` option of `Spider`
- Increased `max-attributes` in `.pylintrc`
  - This should only be a temporary fix until #46 is addressed

### CODE REVIEW

File: tests/test_crawler.py [hunk 1]
```diff
@@ -269,3 +271,160 @@ def test_start_with_save_to_file(
...
+@patch('urllib.request.urlopen')
+def test_respect_robots_txt_not_allowed(mock_urlopen, caplog) -> None: # type: ignore
+    setup_mock_response(
+        url="http://notcrawlable.com",
+        body="<html><body><a href='http://crawlable.com'>link</a></body></html>",
+        status=200
+    )
+
+    mock_urlopen.side_effect = lambda url: (
+        BytesIO(b"User-agent: *\nDisallow: /") if url == "http://notcrawlable.com/robots.txt" else
```
Comment (indrajithi) [inline tests/test_crawler.py:347]: Should we make this into two separate tests and avoid this branch?

Comment (Mews) [reply tests/test_crawler.py:347]: What do you mean?

File: tests/networking/test_robots_txt.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+from unittest.mock import patch, MagicMock
+from io import BytesIO
+import urllib.robotparser
+
+from tiny_web_crawler.networking.robots_txt import get_robots_txt_url, is_robots_txt_allowed, setup_robots_txt_parser
+
+def test_get_robots_txt_url() -> None:
```
[the remark this answers is not in the document]
Comment (Mews) [reply tests/networking/test_robots_txt.py:7]: Alright 👍

[the remark this answers is not in the document]
Comment (Mews) [reply tests/networking/test_robots_txt.py:7]: Done 👍

[the remark this answers is not in the document]
Comment (Mews) [reply tests/networking/test_robots_txt.py]: I also added parametrize to `test_format_url`

**Pull request merged**